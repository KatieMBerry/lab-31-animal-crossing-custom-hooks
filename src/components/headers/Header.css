.Header {
    width: 100%;
}

h1 {
    font-family: 'Courier New', Courier, monospace;
    font-size: x-large;
    color: teal;
}

ul {
    list-style: none;
    font-family: 'Courier New', Courier, monospace;
    font-size: large;
    color: teal;
}

a {
    color: teal;
    font-size: 25px;
    font-weight: bold;
    text-decoration: none;
}
