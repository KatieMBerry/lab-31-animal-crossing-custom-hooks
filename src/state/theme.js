import React, { useState, createContext, useContext } from 'react';

export const ThematicContext = createContext(null);

//provides state to child components
export const ThemeProvider = ({ children }) => {
    const [theme, setTheme] = useState('dark');

    const toggle = ({ target }) => {
        if (target.checked) setTheme('light')
        else setTheme('dark')
    };

    return (//sets up the provider and the values available/provided to children
        <ThematicContext.Provider value={{ theme, useToggle }}>
            {children}
        </ThematicContext.Provider>
    );
};

export const useToggle = () => {
    const { theme } = useContext(ThematicContext);
    return theme;
}

// export const useToggleOn = () => {
//     const { theme } = useContext(ThematicContext);
//     return toggle;
// }
