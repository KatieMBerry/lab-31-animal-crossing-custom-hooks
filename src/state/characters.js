import { useState, useEffect } from 'react';
import { getVillagerById, getVillagers } from '../services/getVillagers';
import { useToggle } from './theme';

export const useVillagers = () => {
    const [loading, setLoading] = useState(true);
    const [villagers, setVillagers] = useState([]);

    useEffect(() => {
        getVillagers()
            .then(villagers => {
                setVillagers(villagers);
                setLoading(false);
            });
    }, []);

    return {
        loading,
        villagers
    }
};

export const useVillagerById = _id => {
    const [loading, setLoading] = useState(true);
    const [villager, setVillager] = useState([]);
    // const toggleOn = useToggle();

    useEffect(() => {
        getVillagerById(_id)
            .then(villager => {
                setVillager(villager);
                setLoading(false);
            });
    }, [_id]);

    return {
        loading,
        villager
    }
}

