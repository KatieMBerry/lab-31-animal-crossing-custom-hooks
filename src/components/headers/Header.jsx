import React, { useContext } from 'react';
import { ThematicContext, useToggleTheme } from '../../state/theme';
import { Link } from 'react-router-dom';
import styles from './Header.css';

const Header = () => {
    //header asks provider for:
    const { toggleTheme } = useContext(ThematicContext);
    // const toggleTheme = useToggleTheme();

    return <div className={`${styles.Header}`}>
        <h1>Welcome to Animal Crossing Api!  Pick Your Villager!</h1>
        <button onClick={toggleTheme}>✨Toggle Mode✨</button>

        <ul>
            <li>
                <Link to="/">Villager Home</Link>
            </li>
        </ul>
    </div>

};

export default Header;
