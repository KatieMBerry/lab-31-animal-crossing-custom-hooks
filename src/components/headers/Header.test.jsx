import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import { render, cleanup } from '@testing-library/react';
import Header from './Header';
import { ThemeProvider } from '../../state/theme';

describe('Header component', () => {
    afterEach(() => cleanup());
    it('renders Header', () => {
        const { asFragment } = render(<Router><ThemeProvider><Header /></ThemeProvider></Router>);
        expect(asFragment()).toMatchSnapshot();
    });
});
