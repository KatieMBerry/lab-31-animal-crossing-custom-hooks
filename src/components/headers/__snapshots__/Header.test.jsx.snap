// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component renders Header 1`] = `
<DocumentFragment>
  <div
    class="Header"
  >
    <h1>
      Welcome to Animal Crossing Api!  Pick Your Villager!
    </h1>
    <button>
      ✨Toggle Mode✨
    </button>
    <ul>
      <li>
        <a
          href="/"
        >
          Villager Home
        </a>
      </li>
    </ul>
  </div>
</DocumentFragment>
`;
